src/services/codefixes/fixMissingTypeAnnotationOnExports.ts

```diff
@@ -392,7 +392,7 @@ function withContext<T>(
                 sourceFile,
                 targetNode,
                 createSatisfiesAsExpression(
-                    getSynthesizedDeepClone(targetNode),

```

Comment:
The `formattingScanner` requires the Identifier to have `parent` for scanning attributes with a dash (data-foo). When we clone the node, we lose the `parent` node, which causes issues. I think in this case, we can avoid deep cloning. Are there any cases where deep cloning is necessary? @jakebailey

Reply:
I think deep cloning is always the correct thing to do here, because we're creating new AST stuff and don't want to reuse old nodes. Can we just `setParentRecursive`?

Reply:
I actually just checked if we could add `return setParentRecursive(clone, false);` to `getSynthesizedDeepClone` itself and it acutally worked, actually _improving_ another test's output. Makes me wonder if that's a better fix (at the cost of us doing this more).

Reply:
@andrewbranch Do you have any thoughts about this?

Reply:
Seems fine to me?

Reply:
Ah, just noticed the PR was changed to set parents for all cloned nodes; that's what I was askng about, yeah. (though tests need to be updated)